src/map.cpp

```diff
@@ -228,12 +228,12 @@ void Map::addNodeAndUpdate(v3s16 p, MapNode n,
 		std::vector<std::pair<v3s16, MapNode> > oldnodes;
 		oldnodes.emplace_back(p, oldnode);
 		voxalgo::update_lighting_nodes(this, oldnodes, modified_blocks);
-
-		for (auto &modified_block : modified_blocks) {
-			modified_block.second->expireDayNightDiff();
-		}
 	}
 
+	if (n.getContent() != oldnode.getContent() &&
+			(oldnode.getContent() == CONTENT_AIR || n.getContent() == CONTENT_AIR))
+		block->expireIsAirCache();
```

Comment:
This wasn't actually correct before but vmanips bypass this code path.
I was going to suggest moving it to `MapBlock::raiseModified` but that removes the ability to do the expiry with a granularity like here.

so just add it to `MMVManip::blitBackAll` too